### ISSUE TITLE
[Issue]: Artist with "²" in the name not handled correctly

### ISSUE THREAD

**Issue by TastyPi** [jellyfin/jellyfin#9725]
### Please describe your bug

I have an artist in my music library called 2winz², however it gets imported into Jellyfin as 2winz�.

To make that clear, here is the hex of those strings:

2winz²
```
0x32 0x77 0x69 0x6e 0x7a 0xc2 0xb2
```

2winz�:
```
0x32 0x77 0x69 0x6e 0x7a 0xef 0xbf 0xbd
```

If I try to rename the artist's metadata title to the correct version, it stops being linked to the album.

It appears at some point Jellyfin is en/decoding the artist name incorrectly.

### Jellyfin Version

10.8.z

### Environment

```markdown
- OS: Arch Linux
- Linux Kernel: 6.2.13-arch1-1
- Virtualization: Docker (lscr version)
- Clients: Browser
- Browser: Firefox 112.0.2
- FFmpeg Version: N/A
- Playback Method: N/A
- Hardware Acceleration: N/A
- GPU Model: N/A
- Plugins: N/A
- Reverse Proxy: N/A
- Base URL: N/A
- Networking: N/A
- Storage: local btrfs
```

### Jellyfin logs

```shell
I could not find anything relevant.
```

### FFmpeg logs

```shell
N/A
```

### Please attach any browser or client logs here

N/A

### Please attach any screenshots here

Path of the album:

![image](https://user-images.githubusercontent.com/934844/235832999-40e8a35a-832f-4b73-a9d3-7b71fd991b7b.png)

(the file's metadata also sets 2winz² as the artist.)

Corrupted artist name on the album's page:

![image](https://user-images.githubusercontent.com/934844/235833093-1ad54cc8-208b-4a74-8b44-d1902af90d22.png)

Corrupted artist's metadata:

![image](https://user-images.githubusercontent.com/934844/235833142-4b985754-ca32-4a5b-8154-fa70b85b1ddd.png)

After changing the artist's title to 2winz² the album is unlinked:

![image](https://user-images.githubusercontent.com/934844/235833204-9c0ae8c4-cf05-415e-b13c-ad64a07dd67d.png)

(The "more like this" is linking to an artist with the path `/media/music/2winz²`, i.e. the correct artist, but neither that artist nor the originally corrupted artist display the album on their pages)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by TastyPi** [comment]
This is still broken in 10.8.10

**Comment by senpai-notices** [comment]
I'm unable to repro this on macOS. `2winz²` displays as expected.

**Comment by TastyPi** [comment]
I decided to dig into this to fix it.

First I tried renaming the directory to not include the unicode character, and that didn't fix it.

Then I decided to dig around Jellyfin's files for any clues and stumbled across this:

![Screenshot from 2024-02-06 01-30-54](https://github.com/jellyfin/jellyfin/assets/934844/4e99ec3b-b02d-4f3e-a511-989b101f18c9)

This was under `metadata/artists`. Note that the "(invalid encoding)" is not actually part of the name, it's added there by my file manager. So that's a pretty big clue that there's an encoding mismatch.

I then decided to try and replicate it by adding the artist name to a public domain file and... it imported correctly. Huh.

Finally, I decided to have a closer look at my original file for any clues, maybe it was secretly UTF-16 when it should be UTF-8 or something. Then I found the real cause - it had a `Script` tag with a value of "Latn". Which I thiiink is supposed to mean that the text in the tags should be treated as ASCII? Not 100% on that part, all I know is deleting that tag made importing work correctly.

TL;DR remove the `Script` tags from your music files.
